### ISSUE TITLE
Questions on fine-tuning freeze and config parameters

### ISSUE THREAD

**Issue by tazalapizza** [YvanYin/Metric3D#173]
Hello, thanks for sharing this amazing work with all the codes and weights !
I created my own RGB-D dataset with custom dataloaders following #105 and ran a finetuning

My finetuning results are not very good with base KITTI parameters, so I want to better understand them :

1) I added encoder freeze, is it a good idea ?

2) how should I choose crop_size values in data_basic ? I set it to my image size

3) the optimizer and lr hyperparameters in ..kitti.py configs are set for finetuning, right ?

4) should we change the Normalize values in the pipeline for custom data ?

5) is there a special way to handle the sky region in GT depth ? I set it to 0 to ignore it

Sorry for my many questions, I hope you can help me. Thank you !

**Comment by JUGGHM** [comment]
> Hello, thanks for sharing this amazing work with all the codes and weights ! I created my own RGB-D dataset with custom dataloaders following #105 and ran a finetuning
> 
> My finetuning results are not very good with base KITTI parameters, so I want to better understand them :
> 
> 1. I added encoder freeze, is it a good idea ?
> 2. how should I choose crop_size values in data_basic ? I set it to my image size
> 3. the optimizer and lr hyperparameters in ..kitti.py configs are set for finetuning, right ?
> 4. should we change the Normalize values in the pipeline for custom data ?
> 5. is there a special way to handle the sky region in GT depth ? I set it to 0 to ignore it
> 
> Sorry for my many questions, I hope you can help me. Thank you !

Thanks for your questions and I hope these may help:
1. It will impair some performance. (The performance gain will shrink for fine-tuning.)

2. I think we have provided a [json file](https://github.com/YvanYin/Metric3D/blob/34afafe58d9543f13c01b65222255dab53333838/training/kitti_json_files/eigen_train.json) to train KITTI and you do not need to change or choose it.

3. No, this file is not correctly configured. I think the settings have been correctly [overridden](https://github.com/YvanYin/Metric3D/blob/34afafe58d9543f13c01b65222255dab53333838/training/mono/configs/RAFTDecoder/vit.raft5.large.kitti.py) here.

4. Not likely.

5. If you have a well segmented map, you can set the sky groundtruth to 200. Ignoring them is another alternative and the confidence map will help to filter them out.

In general, I think we should use this [script](https://github.com/YvanYin/Metric3D/blob/34afafe58d9543f13c01b65222255dab53333838/training/mono/scripts/train_scripts/train_kitti.sh). Not need to change any settings additionally.

**Comment by RNSTD** [comment]
> Hello, thanks for sharing this amazing work with all the codes and weights ! I created my own RGB-D dataset with custom dataloaders following [#105](https://github.com/YvanYin/Metric3D/issues/105) and ran a finetuning
> 
> My finetuning results are not very good with base KITTI parameters, so I want to better understand them :
> 
> 1. I added encoder freeze, is it a good idea ?
> 2. how should I choose crop_size values in data_basic ? I set it to my image size
> 3. the optimizer and lr hyperparameters in ..kitti.py configs are set for finetuning, right ?
> 4. should we change the Normalize values in the pipeline for custom data ?
> 5. is there a special way to handle the sky region in GT depth ? I set it to 0 to ignore it
> 
> Sorry for my many questions, I hope you can help me. Thank you !

Hello, I also want to fine-tune on the KITTI dataset, but the training process is not going smoothly and I'm encountering many errors. Could you please share the code for fine-tuning on KITTI? My email is xtsgdsl@163.com. Thank you!
